parallel_cluster/bin/start_cluster.sh

```diff
@@ -1,21 +1,87 @@
-#!/bin/sh
+#!/bin/bash
+
+# Functions
+echo_stderr(){
+  # Write log to stderr
+  echo "${@}" 1>&2
+}
 
 display_help() {
-    echo "Usage: $0 NAME_OF_YOUR_CLUSTER" >&2
-    exit 1
+  # Display help message then exit
+  echo_stderr "Usage: $0 --cluster-name NAME_OF_YOUR_CLUSTER --cluster-template CLUSTER_TEMPLATE"
 }
 
-if [ "$1" ] ; then
-    pcluster create $1 --config conf/config --cluster-template tothill --norollback 
-    aws ec2 describe-instances \
-             --query "Reservations[*].Instances[*].[InstanceId]" \
-             --filters "Name=instance-state-name,Values=running" "Name=tag:Name,Values=Master" --output text
-
-	# XXX: control error codes better, avoiding counterintuitive ones: i.e authed within a different account:
-	# ERROR: The configuration parameter 'vpc_id' generated the following errors:
-	# The vpc ID 'vpc-7d2b2e1a' does not exist
-    echo "ssm i-XXXXXXXXX away!"
-    exit 0
+# Get args
+while [ $# -gt 0 ]; do
+  echo $1
+  case "$1" in
+    --cluster-name)
+      cluster_name_arg="$2"
+      shift 1
+      ;;
+    --cluster-template)
+      cluster_template_arg="$2"
+      shift 1
+      ;;
+    --config-file)
+      config_file_arg="$2"
+      shift 1
+      ;;
+    --extra-parameters)
+      extra_parameters_arg="$2"
+      shift 1
+      ;;
+    --help)
+      display_help
+      exit 0
+      ;;
+    *)
```

Comment:
We can have the cake and it it too: If it doesn't begin with `--`, assume it's a clustername and Bob's your uncle?